File: src/Defender.BudgetTracker/src/Domain/Entities/RegularExpenses/RegularExpense.cs
using Defender.BudgetTracker.Domain.Entities.Interfaces;
using Defender.BudgetTracker.Domain.Enums;
using Defender.Common.Entities;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Defender.BudgetTracker.Domain.Entities.RegularExpenses;

public class RegularExpense : IUserOwnedModel, IBaseModel
{
    [BsonId]
    [BsonRepresentation(BsonType.String)]
    public Guid Id { get; set; }

    [BsonGuidRepresentation(GuidRepresentation.CSharpLegacy)]
    public Guid UserId { get; set; }

    public string Name { get; set; } = string.Empty;

    [BsonRepresentation(BsonType.String)]
    public RegularExpenseType Type { get; set; }

    [BsonRepresentation(BsonType.String)]
    public Currency Currency { get; set; }

    public long DefaultAmount { get; set; }

    public int OrderPriority { get; set; }
}

File: src/Defender.Portal/src/WebUI/ClientApp/src/models/budgetTracker/regularExpenses/RegularExpense.ts
import { Currency } from "src/models/shared/Currency";

import { RegularExpenseType } from "./RegularExpenseType";

export interface RegularExpense {
  id: string;
  userId?: string;
  name: string;
  type: RegularExpenseType;
  currency: Currency;
  defaultAmount: number;
  orderPriority: number;
}

export interface RegularExpenseInput {
  name: string;
  type: RegularExpenseType;
  currency: Currency;
  defaultAmount: number;
  orderPriority: number;
}
